go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -336,27 +450,24 @@ func runOvnKube(ctx *cli.Context, cancel context.CancelFunc) error {
 			return fmt.Errorf("error when trying to initialize libovsdb SB client: %v", err)
 		}
 
-		// register prometheus metrics that do not depend on becoming ovnkube-master leader
-		metrics.RegisterMasterBase()
-
-		masterEventRecorder = util.EventRecorder(ovnClientset.KubeClient)
-		cm := controllerManager.NewNetworkControllerManager(ovnClientset, getNetworkControllerManagerIdentity(ctx),
-			masterWatchFactory, libovsdbOvnNBClient, libovsdbOvnSBClient, masterEventRecorder, wg)
-		err = cm.Start(ctx.Context, cancel)
-		defer cm.Stop()
+		cm := controllerManager.NewNetworkControllerManager(ovnClientset, runMode.identity,
+			masterWatchFactory, libovsdbOvnNBClient, libovsdbOvnSBClient, eventRecorder, wg)
+		err = cm.Start(ctx)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to start network manager: %w", err)
```

Comment:
I actually used `network manager` on purpose because I don't like `network controller manager` (as I see `controller` as an implementation detail) but let's stick to the name